nexusutils/dataconverter/writer.py

```diff
@@ -244,15 +254,14 @@ def write(self):
                         else:
                             hdf5_links_for_later.append([data, grp, entry_name, self.output_path])
                     else:
-                        dataset = grp.create_dataset(entry_name,
-                                                     data=data
-                                                     )
+                        dataset = grp.create_dataset(entry_name, data=data)
                     units_key = f"{path}/@units"
                     if units_key in self.data.keys() and self.data[units_key] is not None:
                         dataset.attrs["units"] = self.data[units_key]
                     else:
                         continue
                 else:
+                    # consider changing the name here the lvalue can also be group!
```

Comment:
It won't be a group due to the internal ordering of the template dict. But the return type of the function can be updated to reflect this.